src/platform/cmx/boot_comp.c

```diff
@@ -0,0 +1,424 @@
+#include "assert.h"
+#include "kernel.h"
+#include "boot_comp.h"
+#include "chal_cpu.h"
+#include "mem_layout.h"
+#include "string.h"
+#include <pgtbl.h>
+#include <thd.h>
+#include <component.h>
+#include <inv.h>
+#include <hw.h>
+
+#define CMX_FRT1_FLASH 0x08040000
+#define CMX_FRT2_FLASH 0x08060000
+#define CMX_FRT1_RAM   0x20050000
+#define CMX_FRT2_RAM   0x20060000
+
+/* These are used to setup components */
+struct cmx_comp_header
+{
+	unsigned int* data_load;
+	unsigned int* data_start;
+	unsigned int* data_end;
+	unsigned int* bss_start;
+	unsigned int* bss_end;
+	/* The entry of the stack(not the actual function!!) */
+	unsigned int* cos_upcall_stack;
+	/* The entry of the sinv stack(not the actual function!!) */
+	unsigned int* cos_sinv_stack;
+};
+
+void*
+cos_cmx_loadcomp(struct cmx_comp_header* header)
+{
+	u32_t* src;
+	u32_t* ptr;
+	/* data section initialization */
+	src=header->data_load;
+	for(ptr=header->data_start;ptr<header->data_end;ptr++,src++)
```

Comment:
spacing in loop construct needs attention. 
`for(ptr = header->data_start; ptr < header->data_end; ptr++, src++)`